Source/Core/DolphinQt2/CheatsManager.cpp

```diff
@@ -482,11 +482,17 @@ void CheatsManager::Update()
     return;
   }
 
-  m_updating = true;
-
   m_result_label->setText(tr("%1 Match(es)").arg(m_results.size()));
   m_match_table->setRowCount(static_cast<int>(m_results.size()));
 
+  if (m_results.empty())
+    return;
+
+  m_updating = true;
+
+  const auto prev_state = Core::GetState();
```

Comment:
As in: `Core::State prev_state; Core::RunAsCPUThread({ prev_state =; Set State ( paused);  }); `

Reply:
RunAsCPUThread should automatically pause emulation and resume after the function returns, so it would just be:

    Core::RunAsCPUThread([] {
        for (size_t i = 0; i < m_results.size(); i++) ...
    });